src/modules/comments/comments.service.ts

```diff
@@ -0,0 +1,50 @@
+import { UsersService } from "modules/users/users.service";
+import { CommentsRepository } from "./comments.repository";
+import { Comment } from "shared/entities";
+import { PostsService } from "modules/posts/posts.service";
+import { CommentsCreateDto } from "./dto";
+import { v4 as uuid } from "uuid";
+
+export class CommentsService {
+    static async getCommentsByPostId(postId: string): Promise<Comment[]> {
+        const post = await PostsService.getPost(postId);
+
+        return CommentsRepository.getCommentsByPostId(post.id);
+    }
+
+    static async getComment(commentId: string): Promise<Comment> {
```

Comment:
why do you need this function ? Do you really need to get comment by id ?

Reply:
I get to use it when handling the `[PATCH /comments/:commentId]` route inside the user authorization middleware. To check that the comment actually belongs to the user I have to lookup its `user_id` property and match it against the authenticated user's `id`. I then use the getComment (by id) method to fetch the comment itself.
```TypeScript
    // authorizeUser.middleware.ts
    var entity: BlogPost | Comment;
    var field: "author_id" | "user_id";
    if (...) {
        ...
    } else if (commentId) {
        entity = await CommentsService.getComment(commentId);
        field = "user_id";
    }
    const user = await UsersService.getUserById(res.locals.userId);

    try {
        if (user.id != entity[field]) {
        ...
```